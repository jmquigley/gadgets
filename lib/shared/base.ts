import * as React from 'react';
import {Size} from './index';

const styles = require('./styles.css');

export interface BaseOptions {
	className?: boolean;
	disabled?: boolean;
	visible?: boolean;
}

const defaultBaseOptions: BaseOptions = {
	className: true,
	disabled: true,
	visible: true
};

/**
 * A base class between React and all components in the module.
 */
export abstract class BaseComponent<P, S> extends React.Component<P, S> {

	protected _classes: string[] = [];
	protected _inlineStyle: any = {};    // inline style overrides
	protected _styles: any = {};         // css modules styles

	constructor(props: P, pstyles: any = {}) {
		super(props);
		this._styles = pstyles;
	}

	get classes(): string[] {
		return this._classes;
	}

	set classes(str: string[]) {
		this._classes = str;
	}

	get inlineStyle(): any {
		return this._inlineStyle;
	}

	set inlineStyle(val: any) {
		this._inlineStyle = Object.assign(this._inlineStyle, val);
	}

	get style(): string {
		return this._inlineStyle;
	}

	get styles(): any {
		return this._styles;
	}

	protected buildStyles(props: P, style: any = {}, opts?: BaseOptions): void {
		this._classes = [];
		this._inlineStyle = Object.assign(this._inlineStyle, props['style'], style);

		opts = Object.assign(
			defaultBaseOptions,
			opts
		);

		if (props['className'] !== '' && opts.className) {
			this._classes.push(props['className']);
		}

		if (!props['visible'] && opts.visible) {
			this._classes.push(this.styles.invisible);
		}

		if (props['disabled'] && opts.disabled) {
			this._classes.push(styles.disabled);
			this._classes.push('nohover');
		}
	}

	/**
	 * Takes a sizing parameter and a styles object and searches for the
	 * corresponding style class CSS.  If it is found, then it returns the local
	 * style for that size.
	 * @returns {string} the name of the local style for that size.  If
	 * it is not found, then an empty string is returned.
	 */
	protected getSizeStyle() {

		switch (this.props['size']) {
		case Size.xxsmall:
			if ('xxsmall' in this.styles) {
				return this.styles.xxsmall;
			}
			break;

		case Size.xsmall:
			if ('xsmall' in this.styles) {
				return this.styles.xsmall;
			}
			break;

		case Size.small:
			if ('small' in this.styles) {
				return this.styles.small;
			}
			break;

		case Size.large:
			if ('large' in this.styles) {
				return this.styles.large;
			}
			break;

		case Size.xlarge:
			if ('xlarge' in this.styles) {
				return this.styles.xlarge;
			}
			break;

		case Size.xxlarge:
			if ('xxlarge' in this.styles) {
				return this.styles.xxlarge;
			}
			break;

		case Size.normal:
		case Size.medium:
		default:
			if ('medium' in this.styles) {
				return this.styles.medium;
			}
		}

		return '';
	}
}
