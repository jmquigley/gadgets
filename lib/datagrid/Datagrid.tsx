/**
 * A component that represents data in a 2D table format like excel.  The
 * underlying code is a wrapper around the [react-data-grid](https://www.npmjs.com/package/react-data-grid)
 * component.  The wrapper provides themeing and interop with the rest of the component library.
 *
 * ## Screen:
 * <img src="https://github.com/jmquigley/gadgets/blob/master/images/datagrid.png" width="60%" />
 *
 * #### Examples:
 *
 * ```javascript
 * this.state = {
 *     columns: [
 *         {key: "id", name: "ID", sortable: true},
 *         {key: "title", name: "Title"},
 *         {key: "count", name: "Count"}
 *     ],
 *     rows: [
 *         {id: 0, title: "row1", count: 20},
 *         {id: 1, title: "row2", count: 234},
 *         {id: 2, title: "row3", count: 2255},
 *         {id: 3, title: "row4", count: 8345}
 *     ]
 * };
 *
 * ...
 *
 * <Datagrid
 *     columns={this.state.columns}
 *     rows={this.state.rows}
 *     disabled={this.props["disabled"]}
 *     sizing={this.props["sizing"]}
 * />
 * ```
 *
 * ## API
 * For a list of events and atributes for the component see the
 * [react-data-grid](http://adazzle.github.io/react-data-grid/docs/ReactDataGrid)
 * component online help.  This component passes through attributes to the
 * underlying control.
 *
 * #### Styles
 * - `ui-datagrid` - Root style applied to the wrapper around the data grid control
 *
 * #### Properties
 * - `columns=[] {DatagridColumn[]}` - the column descriptor array.  There is
 * on entry in the array per column.
 * - `rows=[] {DatagridRow[]}` - the rows that will be displayed.  Each array
 * entry is an object that maps 1 for 1 to the keys names in the columns
 * descriptor array.
 *
 * @module Datagrid
 */

// const debug = require("debug")("gadgets.Datagrid");
const debugCreate = require("debug")("gadgets.Datagrid:create");
const debugRender = require("debug")("gadgets.Datagrid:render");

import autobind from "autobind-decorator";
import * as React from "react";
import ReactDataGrid from "react-data-grid";
import styled from "styled-components";
import {
	BaseComponent,
	BaseProps,
	BaseState,
	Color,
	disabled,
	fontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	invisible,
	Sizing,
	Wrapper
} from "../shared";

export interface DatagridColumn {
	key: string;
	name: string;
	[key: string]: any;
}

export interface DatagridRow {
	[key: string]: any;
}

export interface DatagridProps extends BaseProps {
	columns: DatagridColumn[];
	rows: DatagridRow[];
	[key: string]: any;
}

export function getDefaultDatagridProps(): DatagridProps {
	return {
		...getDefaultBaseProps(),
		columns: [],
		rows: [],
		obj: "Datagrid"
	};
}

export type DatagridState = BaseState;

export function getDefaultDatagridState(): DatagridState {
	return {...getDefaultBaseState()};
}

const DatagridWrapper: any = styled.div`
	.react-grid-HeaderRow {
		background-color: ${(props: DatagridProps) =>
			props.theme.headerBackgroundColor || Color.black} !important;
		color: ${(props: DatagridProps) =>
			props.theme.headerForegroundColor || Color.white} !important;
		overflow: hidden;
	}

	.react-grid-HeaderCell {
		align-items: center !important;
		background: unset;
		display: inline-flex !important;
		padding: 0 8px;
	}

	${(props: DatagridProps) => fontStyle[props["sizing"]]}
	${(props: DatagridProps) => disabled(props)}
	${(props: DatagridProps) => invisible(props)}
`;

export class Datagrid extends BaseComponent<DatagridProps, DatagridState> {
	public static readonly defaultProps: DatagridProps = getDefaultDatagridProps();
	public state: DatagridState = getDefaultDatagridState();

	constructor(props: DatagridProps) {
		super(props, "ui-datagrid", Datagrid.defaultProps.style);
		debugCreate("props: %O, state: %O", this.props, this.state);
	}

	/** @return the height of the row in pixels (number) */
	get rowHeight() {
		switch (this.props.sizing) {
			case Sizing.xxsmall:
				return 20;
			case Sizing.xsmall:
				return 25;
			case Sizing.small:
				return 30;
			case Sizing.large:
				return 40;
			case Sizing.xlarge:
				return 45;
			case Sizing.xxlarge:
				return 50;

			case Sizing.normal:
			case Sizing.medium:
			default:
				return 35;
		}
	}

	@autobind
	private rowGetter(rowIdx: number) {
		return this.props.rows[rowIdx];
	}

	public render() {
		debugRender("props: %O, state: %O", this.props, this.state);

		const {minHeight, minWidth, ...props} = this.props;

		return (
			<Wrapper {...props}>
				<DatagridWrapper
					{...props}
					className={this.className}
					rowHeight={this.rowHeight}
				>
					<ReactDataGrid
						{...props}
						headerFiltersHeight={this.rowHeight}
						headerRowHeight={this.rowHeight}
						rowGetter={this.rowGetter}
						rowHeight={this.rowHeight}
						rowsCount={props.rows.length}
					/>
				</DatagridWrapper>
			</Wrapper>
		);
	}
}

export default Datagrid;
