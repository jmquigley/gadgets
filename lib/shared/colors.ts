export type ColorType = string;

export enum ColorScheme {
	c1 = "#004358",
	c2 = "#1f8a70",
	c3 = "#bedb39",
	c4 = "#ffe11a",
	c5 = "#fd7400"
}

export enum Color {
	black = "#000000",
	error = "#d9534f",
	glow = "#51cbee",
	gray = "#666666",
	info = "#5bc0de",
	lightgray = "#cccccc",
	lightsaber = "#2ff923",
	red = "#ff0000",
	silver = "#c0c0c0",
	slategray = "#708090",
	success = "#5cb85c",
	ultralightgray = "#e6e6e6",
	warning = "#f0ad4e",
	white = "#ffffff"
}
