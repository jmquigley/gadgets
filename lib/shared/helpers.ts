"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
