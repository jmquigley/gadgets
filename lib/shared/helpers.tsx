'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
