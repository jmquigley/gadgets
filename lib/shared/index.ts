import {
	BaseComponent,
	baseZIndex,
	defaultSize
} from './base';

import {
	getDefaultWrapperProps,
	Wrapper,
	WrapperProps
} from './wrapper';

import {
	BaseProps,
	Color,
	ColorScheme,
	Direction,
	disabled,
	DisabledCSS,
	getDefaultBaseProps,
	invisible,
	InvisibleCSS,
	Justify,
	Location,
	locationStyle,
	SortOrder,
	Styles
} from './props';

import {
	borderStyle,
	boxStyle,
	fontStyle,
	FontStyle,
	lineHeightStyle,
	rectStyle,
	Sizes,
	Sizing,
	Styling
} from './sizing';

import {
	BaseState,
	getDefaultBaseState
} from './state';

import {
	getTheme,
	getThemeList,
	setTheme,
	Theme,
	ThemeProps
} from './themes';

export {
	BaseComponent,
	BaseProps,
	BaseState,
	baseZIndex,
	borderStyle,
	boxStyle,
	Color,
	ColorScheme,
	defaultSize,
	Direction,
	disabled,
	DisabledCSS,
	fontStyle,
	FontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	getDefaultWrapperProps,
	getTheme,
	getThemeList,
	invisible,
	InvisibleCSS,
	Justify,
	lineHeightStyle,
	Location,
	locationStyle,
	rectStyle,
	setTheme,
	Sizes,
	Sizing,
	SortOrder,
	Styles,
	Styling,
	Theme,
	ThemeProps,
	Wrapper,
	WrapperProps
};
