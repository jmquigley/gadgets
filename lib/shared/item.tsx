'use strict';

import {cloneDeep} from 'lodash';
import * as React from 'react';
import {BaseProps, getDefaultBaseProps} from './props';
import {Title, TitleProps} from '../title';

const styles = require('./styles.css');

export interface ItemProps extends BaseProps, TitleProps {
	hiddenLeftButton?: boolean;
	hiddenRightButton?: boolean;
	leftButton?: any;
	rightButton?: any;
	selected?: boolean;
	title?: string;
}

export function getDefaultItemProps(): ItemProps {
	return cloneDeep(Object.assign(
		getDefaultBaseProps(), {
			hiddenLeftButton: false,
			hiddenRightButton: false,
			leftButton: null,
			rightButton: null,
			selected: false,
			stacked: false,
			title: '',
			widget: null
		}));
}

export const ItemComponent = (props: ItemProps) => {

	let leftButton = null;
	if (props.leftButton != null && !props.disabled) {
		leftButton = (
			<div className={`${styles.itemButton} ${(props.hiddenLeftButton) ? styles.hiddenButton : ''}`}>
				{props.leftButton != null ? props.leftButton : null}
			</div>
		);
	}

	let rightButton = null;
	if (props.rightButton != null && !props.disabled) {
		rightButton = (
			<div className={`${styles.itemButton} ${(props.hiddenRightButton) ? styles.hiddenButton : ''}`}>
				{props.rightButton != null ? props.rightButton : null}
			</div>
		);
	}

	return (
		<li
			id={props.id}
			onMouseOut={props.onMouseOut}
			onKeyDown={props.onKeyDown}
			onKeyPress={props.onKeyPress}
			className={`ui-item ${styles.item} ${props.selected ? 'ui-selected' : ''} ${props.classes.join(' ')}`}>
			{leftButton}
			<Title {...props} className={styles.itemTitle} classes={[]}>{props.title}</Title>
			{rightButton}
		</li>
	);
};
