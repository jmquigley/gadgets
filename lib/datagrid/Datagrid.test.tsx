"use strict";

import {EnumValues as ev} from "enum-values";
import {mount, shallow} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {Sizing} from "../shared";
import Datagrid, {DatagridColumn, DatagridRow} from "./Datagrid";

const columns: DatagridColumn[] = [
	{key: "id", name: "ID", sortable: true},
	{key: "title", name: "Title"},
	{key: "count", name: "Count"}
];

const rows: Datagridrow[] = [
	{id: 0, title: "row1", count: 20},
	{id: 1, title: "row2", count: 234},
	{id: 2, title: "row3", count: 2255},
	{id: 3, title: "row4", count: 8345}
];

for (const sizing of ev.getNames(Sizing)) {
	test(`Test creation of a simple Datagrid component (${sizing})`, () => {
		const ctl = mount(
			<Datagrid columns={columns} rows={rows} sizing={Sizing[sizing]} />
		);

		assert(ctl);
		expect(ctl).toMatchSnapshot();
	});
}
