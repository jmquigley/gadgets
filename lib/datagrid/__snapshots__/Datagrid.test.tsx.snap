// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a simple Datagrid component (inherit) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="inherit"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (large) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="large"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (medium) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="medium"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (normal) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (small) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="small"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (xlarge) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="xlarge"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (xsmall) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="xsmall"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (xxlarge) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="xxlarge"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a simple Datagrid component (xxsmall) 1`] = `
<Datagrid
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  columns={
    Array [
      Object {
        "key": "id",
        "name": "ID",
        "sortable": true,
      },
      Object {
        "key": "title",
        "name": "Title",
      },
      Object {
        "key": "count",
        "name": "Count",
      },
    ]
  }
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Datagrid"
  padding={null}
  ripple={false}
  rows={
    Array [
      Object {
        "count": 20,
        "id": 0,
        "title": "row1",
      },
      Object {
        "count": 234,
        "id": 1,
        "title": "row2",
      },
      Object {
        "count": 2255,
        "id": 2,
        "title": "row3",
      },
      Object {
        "count": 8345,
        "id": 3,
        "title": "row4",
      },
    ]
  }
  selected={false}
  sizing="xxsmall"
  style={Object {}}
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test retrieval of Datagrid props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "columns": Array [],
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Datagrid",
  "padding": null,
  "ripple": false,
  "rows": Array [],
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Datagrid props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
}
`;
